parl/algorithms/torch/iql.py

```diff
@@ -0,0 +1,116 @@
+#   Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import torch
+import torch.nn as nn
+import parl
+import torch.nn.functional as F
+from torch.optim.lr_scheduler import CosineAnnealingLR
+
+EXP_ADV_MAX = 100.
+
+
+def asymmetric_l2_loss(u, tau):
+    return torch.mean(torch.abs(tau - (u < 0).float()) * u**2)
+
+
+class IQL(parl.Algorithm):
+    def __init__(self,
```

Comment:
Pls add `check_model_method` and assert types of arguments as follows:
https://github.com/PaddlePaddle/PARL/blob/20c4bb941e54fd4e221db828d0c1a2c1e1f1ae8a/parl/algorithms/torch/cql.py#L57